jekyll/_cci2/language-python.md

```diff
@@ -2,104 +2,266 @@
+to execute bash commands.
+In this example,
+[Pipenv](https://pipenv.readthedocs.io/en/latest/) is used
+to create a virtual environment
+and install Python packages.
+
+```yaml
+version: 2
+jobs:
+  build:
+    # ...
+    steps:
+      - checkout  # checkout source code to working directory
+      - run:
+        command: |  # use pipenv to install dependencies
+          sudo pip install pipenv
+          pipenv install
+```
+
+### Cache Dependencies
+
+To save time between runs,
+consider [caching dependencies or source code]({{ site.baseurl }}/2.0/caching/).
+
+Use the [`save_cache`]({{ site.baseurl }}/2.0/configuration-reference/#save_cache) step
+to cache certain files or directories.
+In this example,
+the virtual environment and installed packages are cached.
+
+Use the [`restore_cache`]({{ site.baseurl }}/2.0/configuration-reference/#restore_cache) step
+to restore cached files or directories.
 
 {% raw %}
 
 ```yaml
-    steps: # a collection of executable commands
-      - checkout # special step to check out source code to the working directory
-      - restore_cache: # restores saved dependency cache if the Branch key template or requirements.txt files have not changed since the previous run
-          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
-      - run: # install and activate virtual environment with pip
-          command: |
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -r requirements.txt
-      - save_cache: # special step to save dependency cache
-          key: deps1-{{ .Branch }}-{{ checksum "requirements.txt" }}
+version: 2
+jobs:
+  build:
+    # ...
+    steps:
+      - checkout
+      - run: sudo chown -R circleci:circleci /usr/local/bin
+      - run: sudo chown -R circleci:circleci /usr/local/lib/python3.6/site-packages
+      - restore_cache:  # ensure this step occurs *before* installing dependencies
+          key: deps9-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+      - run:
+        command: |
+          sudo pip install pipenv
+          pipenv install
+      - save_cache:
+          key: deps9-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
           paths:
-            - "venv"
-      - run: # run tests
-          command: |
-            . venv/bin/activate
-            python manage.py test
-      - store_artifacts: # special step to store test reports as artifacts
-          path: test-reports/
-          destination: tr1
-      # See https://circleci.com/docs/2.0/deployment-integrations/ for deploy examples    
+            - ".venv"
+            - "/usr/local/bin"
+            - "/usr/local/lib/python3.6/site-packages"
 ```
 
 {% endraw %}
 
-For the complete list of CircleCI configuration keys,
-refer to the [Configuring CircleCI]({{ site.baseurl }}/2.0/configuration-reference/) document.
+**Note:**
+Use the `chown` command
+to grant CircleCI access to dependency locations.
 
-## See Also
```

Comment:
add back the see also section at the end of this document (all docs now have this section at the end)

Reply:
Most docs do not have this section at the end.

Reply:
Most do, see 

  https://github.com/circleci/circleci-docs/pull/2645 and  https://github.com/circleci/circleci-docs/pull/2339

add back, thank you!

Reply:
Cool, your previous claim that "all docs now have this section" confused me since I found several examples of docs without the section.

I'm happy to add this section back if you know which documents you would like to link to. The previously-linked documents are already linked elsewhere in the document.

Reply:
Your job is to figure out what see also links should be since you decided to move them elsewhere.